_data-prepper/pipelines/configuration/processors/list-to-map.md

```diff
@@ -0,0 +1,305 @@
+      "value": "val-b1"
+    },
+    {
+      "name": "b",
+      "value": "val-b2"
+    },
+    {
+      "name": "c",
+      "value": "val-c"
+    }
+  ],
+  "a": "val-a",
+  "b": "val-b1",
+  "c": "val-c"
+}
+```
+
+### Example: Maps set to `target`
+
+The following example `pipeline.yaml` shows the `list_to_map` processor when set to a specified target, `mymap`.
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        target: "mymap"
+        value_key: "value"
+        flatten: true
+  sink:
+    - stdout:
+```
+{% include copy.html %}
+
+The generated map appears under the target key.
+
+```json
+{
+  "mylist": [
+    {
+      "name": "a",
+      "value": "val-a"
+    },
+    {
+      "name": "b",
+      "value": "val-b1"
+    },
+    {
+      "name": "b",
+      "value": "val-b2"
+    },
+    {
+      "name": "c",
+      "value": "val-c"
+    }
+  ],
+  "mymap": {
+    "a": "val-a",
+    "b": "val-b1",
+    "c": "val-c"
+  }
+}
+```
+
+### Example: No `value_key` specified
+
+The follow example `pipeline.yaml` shows the `list_to_map` processor with no `value_key` specified. Since `key` is set to the objects `name`, the processor extracts the names of each object to use as keys in the map. 
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        flatten: true
+  sink:
+    - stdout:
+```
+{% include copy.html %}
+
+The values from the generated map appear as original objects from the `.log` source as shown in the following example response:
+
+```json
+{
+  "mylist": [
+    {
+      "name": "a",
+      "value": "val-a"
+    },
+    {
+      "name": "b",
+      "value": "val-b1"
+    },
+    {
+      "name": "b",
+      "value": "val-b2"
+    },
+    {
+      "name": "c",
+      "value": "val-c"
+    }
+  ],
+  "a": {
+    "name": "a",
+    "value": "val-a"
+  },
+  "b": {
+    "name": "b",
+    "value": "val-b1"
+  },
+  "c": {
+    "name": "c",
+    "value": "val-c"
+  }
+}
+```
+
+### Example: `flattened_element` set to `last`
+
+The following example `pipeline.yaml` sets the `flattened_element` to last, therefore flattening the processor output based on each value's last element. 
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - list_to_map:
+        key: "name"
+        source: "mylist"
+        target: "mymap"
+        value_key: "value"
+        flatten: true
+        flattened_element: "last"
+  sink:
+    - stdout:
+```
+{% include copy.html %}
+
+The processor maps object `b` to value `val-b2`, since `val-b2` is the last element in object `b`, as shown in the following output:
```

Comment:
```suggestion
The processor maps object `b` to value `val-b2` because `val-b2` is the last element in object `b`, as shown in the following output:
```

Reply:
Another instance of since here. I'll add that to my notes of what to watch our for :)